data/cache/impl.py

```diff
@@ -213,3 +213,72 @@ def retrieve(self, cache_key, loader, should_cache=is_not_none):
             logger.debug("Not caching loaded result for key %s: %s", cache_key.key, result)
 
         return result
+
+
+class RedisDataModelCache(DataModelCache):
+    """
+    Implementation of the data model cache backed by a Redis service.
+    """
+
+    def __init__(self, endpoint):
+        self.host = endpoint[0]
+        self.port = endpoint[1]
+        self.client = None
+
+    def _get_client(self):
+        if self.client is not None:
+            return self.client
+
+        args = {
+            "host": self.host,
+            "port": self.port,
+            "socket_connect_timeout": 1,
+            "socket_timeout": 2,
+            "single_connection_client": True,
+        }
+        self.client = redis.StrictRedis(**args)
+
+        return self.client
+
+    def retrieve(self, cache_key, loader, should_cache=is_not_none):
+        # TODO(alecmerdler): We might want to have different behavior based on `cache_key` (using "sets" for `/tags/list`, single value for others...)
+        not_found = None
+        client = self._get_client()
+        if client is not None:
+            logger.debug("Checking cache for key %s", cache_key.key)
+            try:
+                cached_result = client.get(cache_key.key)
+                if cached_result != not_found:
```

Comment:
Not specific to this implementation, but we could instrument this to measure the hit rate.
e.g A `Counter` metrics for  cache hits divided by a `Counter` metrics for `retrieve` calls